### PR TITLE
chore(api, shared, app-generic): Convert API rate limit FF to be LaunchDarkly compatible

### PR DESCRIPTION
**Pull request by rifont**
### What change does this PR introduce?
* Convert API rate limit flag from system-critical to feature-flag
* Rename `IS_REQUEST_RATE_LIMITING_ENABLED` -> `IS_API_RATE_LIMITING_ENABLED` for naming consistency with features under implementation

### Why was this change needed?
* The feature flag was not compatible with LaunchDarkly and could not be resolved at runtime. This change enables rate limiting to be rolled out gradually through strategies such as A/B.

### Other information (Screenshots)
N/A

### CODE REVIEW

File: apps/api/src/.env.development [hunk 1]
```diff
@@ -25,7 +25,7 @@ REDIS_CACHE_FAMILY=
 REDIS_CACHE_KEY_PREFIX=
 REDIS_CACHE_ENABLE_AUTOPIPELINING=true
 
-IS_REQUEST_RATE_LIMITING_ENABLED=false
+IS_API_RATE_LIMITING_ENABLED=false
```
Comment (rifont) [inline apps/api/src/.env.development:28]: "API Rate Limiting" aligns naming with the use-cases under development. If we decide to add Web Socket rate limiting or other types, we can then toggle these features independently.

File: packages/application-generic/src/usecases/get-feature-flag/get-is-request-rate-limiting-enabled.use-case.ts [hunk 2]
```diff
@@ -1,21 +1,22 @@
...
 
-import { GetSystemCriticalFlag } from './get-system-critical-flag.use-case';
+import { FeatureFlagCommand } from './get-feature-flag.command';
+import { GetFeatureFlag } from './get-feature-flag.use-case';
 
 @Injectable()
-export class GetIsRequestRateLimitingEnabled extends GetSystemCriticalFlag {
-  execute(): boolean {
-    const value = process.env.IS_REQUEST_RATE_LIMITING_ENABLED;
+export class GetIsApiRateLimitingEnabled extends GetFeatureFlag {
```
Comment (rifont) [inline packages/application-generic/src/usecases/get-feature-flag/get-is-request-rate-limiting-enabled.use-case.ts:8]: Changing `GetFeatureFlag` -> `GetSystemCriticalFlag` and modifying the `execute` method accordingly.

File: packages/application-generic/src/usecases/get-feature-flag/get-is-api-rate-limiting-enabled.use-case.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common';
```
Comment (rifont) [inline packages/application-generic/src/usecases/get-feature-flag/get-is-api-rate-limiting-enabled.use-case.ts:1]: This file was also renamed from `get-is-request-rate-limiting-enabled` -> `get-is-api-rate-limiting-enabled`

**Review by LetItRock — APPROVED** [review]
looks good to me 🎉

**Pull request merged**